src/main/java/com/uid2/admin/store/writer/KeysetKeyStoreWriter.java

```diff
@@ -9,21 +9,31 @@
 import com.uid2.shared.store.scope.StoreScope;
 import io.vertx.core.json.JsonArray;
 import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.Base64;
 import java.util.Collection;
 
 public class KeysetKeyStoreWriter implements StoreWriter<Collection<KeysetKey>> {
     private final ScopedStoreWriter writer;
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeysetStoreWriter.class);
+    private final boolean enableKeysets;
 
-    public KeysetKeyStoreWriter(RotatingKeysetKeyStore provider, FileManager fileManager, VersionGenerator versionGenerator, Clock clock, StoreScope scope) {
+    public KeysetKeyStoreWriter(RotatingKeysetKeyStore provider, FileManager fileManager,
+                                VersionGenerator versionGenerator, Clock clock, StoreScope scope, boolean enableKeysets) {
         FileName dataFile = new FileName("keyset_keys", ".json");
         String dataType = "keyset_keys";
         writer = new ScopedStoreWriter(provider, fileManager, versionGenerator, clock, scope, dataFile, dataType);
+        this.enableKeysets = enableKeysets;
     }
 
     @Override
     public void upload(Collection<KeysetKey> data, JsonObject extraMeta) throws Exception {
+        if(!enableKeysets) {
+            LOGGER.error("Uploaded Attempted to Keysets Keys with keysets disabled");
```

Comment:
I am confused about how the PrivateSiteDataSyncJob.java will work. We still execute the keysetSyncJob and keysetKeySyncJob and the writer provided by the factory handles "disabled" but the reader does not. So it will still read, compare, and upload but get stopped at at this line, right? Why even start the sync jobs if keysets are disabled?

Reply:
Ok I added a if blocks to stop the job from getting executed.